src/radical/pilot/agent/executing/popen.py

```diff
@@ -360,6 +360,11 @@ def _handle_task(self, task):
         # store pid for last-effort termination
         _pids.append(task['proc'].pid)
 
+        # keep process ID within metadata in the task description
+        if not task['description'].get('metadata'):
+            task['description']['metadata'] = {}
+        task['description']['metadata']['process_id'] = task['proc'].pid
+
```

Comment:
Note that this PID is the PID of, say, `mpirun`, not of the task process itself.  Is that what Darshan needs?

Otherwise this looks good to me.